test/conf_include_test.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <openssl/conf.h>
+#include <openssl/err.h>
+#include <openssl/ssl.h>
+#include <string.h>
+#include "testutil.h"
+
+#ifdef _WIN32
+# include <direct.h>
+# define DIRSEP "/\\"
+# define chdir _chdir
+#else
+# include <unistd.h>
+# define DIRSEP "/"
+#endif
+
+/* changes path to that of the filename */
+static int change_path(const char *file)
+{
+    char *s = OPENSSL_strdup(file);
+    char *p = s;
+    char *last = NULL;
+    int ret;
+
+    if (s == NULL)
+        return -1;
+
+    while ((p = strpbrk(p, DIRSEP)) != NULL) {
+        last = p++;
+    }
+    if (last == NULL)
+        return 0;
+    *last = 0;
+
+    TEST_note("changing path to %s", s);
+    ret = chdir(s);
+    OPENSSL_free(s);
+    return ret;
+}
+
+/* This test program checks the operation of the .include directive.
+ */
+
+static BIO *dup_bio_in(void)
+{
+    return BIO_new_fp(stdin, BIO_NOCLOSE | BIO_FP_TEXT);
+}
+
+static BIO *dup_bio_out(void)
+{
+    BIO *b = BIO_new_fp(stdout, BIO_NOCLOSE | BIO_FP_TEXT);
+    return b;
+}
+
+static BIO *bio_open_default_(const char *filename, const char *mode, int quiet)
```

Comment:
Here is the thing. As you might remember earlier we discussed test_printf_stderr. One of the ideas behind that arrangement was that test programs should not have direct references to stdio. So that if you find yourself in situation when you need to compile test program on some exotic environment (think bare-metal, no OS, JTAG, ...) you wouldn't have to struggle with stdio, but give it some stubs to link with. Of course one would probably never have to compile *this* test program for bare-metal, because there won't be even a file system, but nevertheless, it's considered undesirable to have references to stdio. So would you consider omitting this?

Reply:
You mean the reference to stdin/stdout in dup_bio_in/out? I can drop that as it is not used by anything in the test.

Reply:
> You mean the reference to stdin/stdout in dup_bio_in/out?

Yes. I've commented at this line, because it's the function in question that calls dup_bio_in/out that in turn refer to [unwanted] stdin/stdout.